Rename `Iterator::array_chunks` to `Iterator::chunks`

rossmacarthur opened the pull request:
The most logical way to yield chunks from an iterator is with arrays. Originally I named this to align with `slice::array_chunks` but I think `chunks` makes more sense for iterators and it aligns with the `next_chunk` method.

Related #100450

rust-log-analyzer commented:
The job **`x86_64-gnu-llvm-13`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/3557368609/jobs/5975329196) [(plain)](https://github.com/rust-lang/rust/commit/c9f42a4d316f1785824491057db01996db42a588/checks/9727316409/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
normalized stderr:
warning: an associated function with this name may be added to the standard library in the future
  --> $DIR/example-calendar.rs:719:31
   |
LL |     let c = r.iter().cloned().chunks(3).collect::<Vec<_>>();
   |
   = warning: once this associated item is added to the standard library, the ambiguity may cause an error or change in behavior!
Some tests failed in compiletest suite=ui mode=ui host=x86_64-unknown-linux-gnu target=x86_64-unknown-linux-gnu
   = note: for more information, see issue #48919 <https://github.com/rust-lang/rust/issues/48919>
   = note: for more information, see issue #48919 <https://github.com/rust-lang/rust/issues/48919>
   = help: call with fully qualified syntax `Chunks::chunks(...)` to keep using the current method
   = help: add `#![feature(iter_array_chunks)]` to the crate attributes to enable `chunks`

warning: an associated function with this name may be added to the standard library in the future
  --> $DIR/example-calendar.rs:734:10
   |
   |
LL |         .chunks(months_per_row)
   |
   = warning: once this associated item is added to the standard library, the ambiguity may cause an error or change in behavior!
   = note: for more information, see issue #48919 <https://github.com/rust-lang/rust/issues/48919>
   = note: for more information, see issue #48919 <https://github.com/rust-lang/rust/issues/48919>
   = help: call with fully qualified syntax `Chunks::chunks(...)` to keep using the current method
   = help: add `#![feature(iter_array_chunks)]` to the crate attributes to enable `chunks`
warning: 2 warnings emitted



---
To only update this specific test, also pass `--test-args impl-trait/example-calendar.rs`

error: 1 errors occurred comparing output.
status: exit status: 0
command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/impl-trait/example-calendar.rs" "-Zthreads=1" "--target=x86_64-unknown-linux-gnu" "-O" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Cstrip=debuginfo" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/impl-trait/example-calendar/a" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/impl-trait/example-calendar/auxiliary"
stdout: none
--- stderr -------------------------------
  --> /checkout/src/test/ui/impl-trait/example-calendar.rs:719:31
   |
   |
LL |     let c = r.iter().cloned().chunks(3).collect::<Vec<_>>();
   |
   = warning: once this associated item is added to the standard library, the ambiguity may cause an error or change in behavior!
   = note: for more information, see issue #48919 <https://github.com/rust-lang/rust/issues/48919>
   = note: for more information, see issue #48919 <https://github.com/rust-lang/rust/issues/48919>
   = help: call with fully qualified syntax `Chunks::chunks(...)` to keep using the current method
   = help: add `#![feature(iter_array_chunks)]` to the crate attributes to enable `chunks`

warning: an associated function with this name may be added to the standard library in the future
  --> /checkout/src/test/ui/impl-trait/example-calendar.rs:734:10
   |
   |
LL |         .chunks(months_per_row)
   |
   = warning: once this associated item is added to the standard library, the ambiguity may cause an error or change in behavior!
   = note: for more information, see issue #48919 <https://github.com/rust-lang/rust/issues/48919>
   = note: for more information, see issue #48919 <https://github.com/rust-lang/rust/issues/48919>
   = help: call with fully qualified syntax `Chunks::chunks(...)` to keep using the current method
   = help: add `#![feature(iter_array_chunks)]` to the crate attributes to enable `chunks`
warning: 2 warnings emitted
------------------------------------------


```

</details>

rust-log-analyzer commented:
The job **`mingw-check`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/3558006251/jobs/5976380317) [(plain)](https://github.com/rust-lang/rust/commit/031f6137a2c24ab32df475e9c4cd572b25deeda4/checks/9728535120/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
configure: rust.debug-assertions := True
configure: rust.overflow-checks := True
configure: llvm.assertions      := True
configure: dist.missing-tools   := True
configure: build.configure-args := ['--enable-sccache', '--disable-manage-submodu ...
configure: writing `config.toml` in current directory
configure: 
configure: run `python /checkout/x.py --help`
Attempting with retry: make prepare
---
skip untracked path cpu-usage.csv during rustfmt invocations
skip untracked path src/doc/book/ during rustfmt invocations
skip untracked path src/doc/rust-by-example/ during rustfmt invocations
skip untracked path src/llvm-project/ during rustfmt invocations
Diff in /checkout/library/core/src/iter/traits/iterator.rs at line 5:
 use super::super::try_process;
 use super::super::ByRefSized;
 use super::super::TrustedRandomAccessNoCoerce;
-use super::super::{Chunks, Chain, Cloned, Copied, Cycle, Enumerate, Filter, FilterMap, Fuse};
+use super::super::{Chain, Chunks, Cloned, Copied, Cycle, Enumerate, Filter, FilterMap, Fuse};
 use super::super::{FlatMap, Flatten};
 use super::super::{FromIterator, Intersperse, IntersperseWith, Product, Sum, Zip};
 use super::super::{
Running `"/checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/rustfmt" "--config-path" "/checkout" "--edition" "2021" "--unstable-features" "--skip-children" "--check" "/checkout/library/core/src/iter/adapters/map_while.rs" "/checkout/library/core/src/iter/traits/iterator.rs" "/checkout/library/core/src/iter/adapters/skip_while.rs" "/checkout/library/core/src/iter/traits/double_ended.rs" "/checkout/library/core/src/iter/traits/marker.rs" "/checkout/library/core/src/iter/adapters/scan.rs" "/checkout/library/core/src/iter/range.rs" "/checkout/library/core/src/iter/traits/collect.rs"` failed.
If you're running `tidy`, try again with `--bless`. Or, if you just want to format code, run `./x.py fmt` instead.
```

</details>

rossmacarthur committed:
Rename `Iterator::array_chunks` to `Iterator::chunks`

The most logical way to yield chunks from an iterator is with arrays.
Originally I named this to align with `slice::array_chunks` but I think
`chunks` makes more sense for iterators and it aligns with the
`next_chunk` method.

jdahlstrom commented:
I would expect `chunks` to be reserved for a method that takes a runtime size and returns an iterator.

rossmacarthur commented:
Yeah, good point 🤔, I wonder if `next_chunk` is better called `next_array`
